### PR TITLE
Server 833

### PR DESCRIPTION
**Pull request by valentinyanakiev**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/cherrytwist/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/cherrytwist/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/cherrytwist/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

File: .env.bckp [hunk 1]
```diff
@@ -0,0 +1,7 @@
+AUTH_AAD_TEST_HARNESS_PASSWORD=Ch3rrytw1$t@0rG
```
Comment (techsmyth) [inline .env.bckp]: is this file needed?

File: src/services/ipfs/ipfs.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    const filePath = `./uploads/${filename}-${guid}`;
+
+    const res = new Promise(async (resolve, reject) =>
+      createReadStream()
+        .pipe(createWriteStream(filePath))
+        .on('finish', () => resolve(true))
+        .on('error', () => reject(false))
+    );
+
+    if (await res) {
```
Comment (techsmyth) [inline src/services/ipfs/ipfs.resolver.ts:42]: remove the temporary file after with ipfs upload succeeds or on an error? Otherwise this temp storage will fill up...

File: src/services/ipfs/ipfs.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { createWriteStream } from 'fs';
+import { FileUpload, GraphQLUpload } from 'graphql-upload';
+import { IpfsService } from './ipfs.service';
+import { v4 as uuidv4 } from 'uuid';
+
+@Resolver()
+export class IpfsResolver {
+  constructor(@Inject(IpfsService) private ipfsService: IpfsService) {}
+
+  @Mutation(() => String)
```
Comment (techsmyth) [inline src/services/ipfs/ipfs.resolver.ts:17]: user should be registered? do not want this world open.

**Review by techsmyth — CHANGES_REQUESTED** [review]
Looks nice!

A few comments to be addressed.

Also important to fix the package security warnings that are identified.

File: .env.bckp (hunk 1 above)
Comment (valentinyanakiev) [reply .env.bckp]: Addressed in 25eb4f5a65de0e22a10bcddbc7317ac9e2360584

File: src/services/ipfs/ipfs.resolver.ts (hunk 2 above)
Comment (valentinyanakiev) [reply src/services/ipfs/ipfs.resolver.ts:42]: Addressed in 25eb4f5a65de0e22a10bcddbc7317ac9e2360584

File: src/services/ipfs/ipfs.resolver.ts (hunk 3 above)
Comment (valentinyanakiev) [reply src/services/ipfs/ipfs.resolver.ts:17]: Addressed in 25eb4f5a65de0e22a10bcddbc7317ac9e2360584

**Review by techsmyth — APPROVED** [review]
